XiEditor/Search.swift

```diff
@@ -0,0 +1,208 @@
+// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+import Swift
+
+class FindViewController: NSViewController, NSTextFieldDelegate {
+
+    @IBOutlet weak var searchField: NSSearchField!
+    @IBOutlet weak var navigationButtons: NSSegmentedControl!
+    @IBOutlet weak var doneButton: NSButton!
+    @IBOutlet weak var ignoreCaseMenuItem: NSMenuItem!
+    @IBOutlet weak var wrapAroundMenuItem: NSMenuItem!
+
+    var optionIgnoreCase: Bool = true
+    var optionWrapAround: Bool = true
+
+    var editViewController: EditViewController!
+
+    override func viewDidLoad() {
+        // add recent searches menu items
+        let menu = searchField.searchMenuTemplate!
+
+        menu.addItem(NSMenuItem.separator())
+
+        let recentTitle = NSMenuItem(title: "Recent Searches", action: nil, keyEquivalent: "")
+        recentTitle.tag = Int(NSSearchFieldRecentsTitleMenuItemTag)
+        menu.addItem(recentTitle)
+
+        let recentItem = NSMenuItem(title: "", action: nil, keyEquivalent: "")
+        recentItem.tag = Int(NSSearchFieldRecentsMenuItemTag)
+        menu.addItem(recentItem)
+
+        menu.addItem(NSMenuItem.separator())
+
+        let recentClear = NSMenuItem(title: "Clear Recent Searches", action: nil, keyEquivalent: "")
+        recentClear.tag = Int(NSSearchFieldClearRecentsMenuItemTag)
+        menu.addItem(recentClear)
+    }
+
+    override func validateMenuItem(_ menuItem: NSMenuItem) -> Bool {
+        return true
+    }
+
+    @IBAction func selectIgnoreCaseMenuItem(_ sender: Any) {
+        if let menuItem = sender as? NSMenuItem {
+            optionIgnoreCase = !optionIgnoreCase
+            if optionIgnoreCase {
+                menuItem.state = NSOnState
+            } else {
+                menuItem.state = NSOffState
+            }
+        }
+
+        editViewController.find(searchField.stringValue,
```

Comment:
I've tested both VSC and Sublime and both re-do the current search when the case-matching is changed. However the EditView should not scoll, if there is no search result. So if this was the case, it is a bug.

A note about this function. I was surprised that I hat to implement these methods at all, because I setup a data binding between the menu items and the `optionXX` properties. However this seems to only effect the initial state of the menu items or I did something wrong.